### PR TITLE
Update solidity-by-example.rst.

### LINKED ISSUE #3750: doc solidity-by-example.rst is out of date
the first example is still used ```pragma solidity ^0.4.16``` as the other examples in the same doc already update to ^0.4.21; 
and in this example, it still have out of date comment for ```require```

```
            // If the argument of `require` evaluates to `false`,
            // it terminates and reverts all changes to
            // the state and to Ether balances. It is often
            // a good idea to use this if functions are
            // called incorrectly. But watch out, this
            // will currently also consume all provided gas
            // (this is planned to change in the future).
            require(
                (msg.sender == chairperson) &&
                !voters[voter].voted &&
                (voters[voter].weight == 0)
            );
```

### PR DESCRIPTION
**Pull request by netrunnerX**
Fixes #3750

### CODE REVIEW

File: docs/solidity-by-example.rst [hunk 1]
```diff
@@ -89,11 +89,10 @@ of votes.
         function giveRightToVote(address voter) public {
             // If the argument of `require` evaluates to `false`,
             // it terminates and reverts all changes to
-            // the state and to Ether balances. It is often
-            // a good idea to use this if functions are
-            // called incorrectly. But watch out, this
-            // will currently also consume all provided gas
-            // (this is planned to change in the future).
+            // the state and to Ether balances. 
+            // It will also refund the remaining gas to the caller.
```
Comment (chriseth) [inline docs/solidity-by-example.rst]: Not sure if we should mention this explicitly, it will not refund gas right away, it will just not consume all gas (which is actually not a special case).

File: docs/solidity-by-example.rst [hunk 2]
```diff
@@ -36,7 +36,7 @@ of votes.
 
 ::
 
-    pragma solidity ^0.4.16;
+    pragma solidity ^0.4.21;
```
Comment (chriseth) [inline docs/solidity-by-example.rst]: Is this change needed?

**Comment by chriseth** [comment]
Please do not include merge commits, but rather use rebase.

File: docs/solidity-by-example.rst (hunk 1 above)
Comment (netrunnerX) [reply docs/solidity-by-example.rst]: So, it's ok if I change the comment to just say require wont consume gas?

Comment (chriseth) [reply docs/solidity-by-example.rst]: Yes!

File: docs/solidity-by-example.rst (hunk 2 above)
Comment (netrunnerX) [reply docs/solidity-by-example.rst]: In the issue they mention that this example was still with ^0.4.16 while the other ones were updated to ^0.4.21. 
If you prefer i can leave it as ^0.4.16.

**Comment by netrunnerX** [comment]
Should I revert the pull request from forked repository and make another pull request with merge rebase?

File: docs/solidity-by-example.rst (hunk 2 above)
Comment (chriseth) [reply docs/solidity-by-example.rst]: If it works with 0.4.16, it should stay at 0.4.16.

**Comment by chriseth** [comment]
Please just force-push to the branch, do not open a new pull request.

File: docs/solidity-by-example.rst [hunk 3]
```diff
@@ -89,11 +89,10 @@ of votes.
         function giveRightToVote(address voter) public {
             // If the argument of `require` evaluates to `false`,
             // it terminates and reverts all changes to
-            // the state and to Ether balances. It is often
-            // a good idea to use this if functions are
-            // called incorrectly. But watch out, this
-            // will currently also consume all provided gas
-            // (this is planned to change in the future).
+            // the state and to Ether balances. 
+            // require wont consume gas.
```
Comment (chriseth) [inline docs/solidity-by-example.rst]: Please use proper English, capitalize the first word in a sentence and and avoid the use of contractions.

Perhaps it is also better to state that this consumes all gas in old EVM versions but not anymore.

Comment (netrunnerX) [reply docs/solidity-by-example.rst]: I'm not used to capitalize a word when it's the name of a function, but you're right, it does not look so good. I update with the suggested statement.

**Comment by chriseth** [comment]
Can you please squash the three commits into a single one and also remove the merge commit?

**Comment by netrunnerX** [comment]
Removed merge commit and squashed commits into a single one.

**Review by chriseth — APPROVED** [review]
